lib/xclarity_client/xclarity_base.rb

```diff
@@ -28,12 +28,13 @@ def connection_builder(conf, uri)
       @conn = Faraday.new(url: url) do |faraday|
         faraday.request  :url_encoded             # form-encode POST params
         faraday.response :logger                  # log requests to STDOUT -- This line, should be uncommented if you wanna inspect the URL Request
+        faraday.use :cookie_jar unless conf.auth_type != 'token'
         faraday.adapter  Faraday.default_adapter  # make requests with Net::HTTP
         faraday.ssl[:verify] = conf.verify_ssl == 'PEER'
       end
 
-      response = authentication(conf) unless conf.auth_type != 'token'
-      #TODO: What's to do with the response of authentication request?
+      authentication(conf) unless conf.auth_type != 'token'
```

Comment:
Use `if conf.auth_type == 'token' ` instead of `unless conf.auth_type != 'token' `